org.mwc.debrief.core/src/org/mwc/debrief/core/ContextOperations/GeneratePasteRepClipboard.java

```diff
@@ -0,0 +1,216 @@
+    return doPasteAction;
+  }
+
+
+  private static class PasteRepOperation extends CMAPOperation{
+
+    private String _contentToImport;
+    private Layers _tempLayers;
+    private Layers _layers;
+    private List<Editable> addedElements;
+    private ImportReplay _tracker;
+    
+    public PasteRepOperation(String title,Layers theLayers,String contentToImport)
+    {
+      super(title);
+      _contentToImport = contentToImport;
+      _layers = theLayers;
+    }
+    @Override
+    public IStatus execute(IProgressMonitor monitor, IAdaptable info)
+        throws ExecutionException
+    {
+      _tracker = new ImportReplay();
+      _tempLayers = new Layers();
+      //import to a temp layers object
+      _tracker.setLayers(_tempLayers);
+      addedElements = _tracker.importThis(_contentToImport,_layers);
+      return Status.OK_STATUS;
+    }
+   
+    @Override
+    public IStatus undo(IProgressMonitor monitor, IAdaptable info)
+        throws ExecutionException
+    {
+      //iterate through the addedElements and see what got added and remove them
+      for(Editable element:addedElements) {
+        if(element instanceof Layer) {
+          _layers.removeThisLayer((Layer)element);
+        }
+        else {
+          Layer layer = null;
+          if(element instanceof DynamicShapeWrapper) {
+            layer = (Layer)_layers.findLayer(((DynamicShapeWrapper)element).getTrackName());
+          }
+          else if (element instanceof SensorContactWrapper)
+          {
+            layer = (Layer)_layers.findLayer(((SensorContactWrapper)element).getTrackName());
+          }
+          else if (element instanceof DynamicTrackShapeWrapper)
+          {
+            layer = (Layer)_layers.findLayer(((DynamicTrackShapeWrapper)element).getTrackName());
+          }
+          else if (element instanceof TMAContactWrapper)
+          {
+            layer = (Layer)_layers.findLayer(((TMAContactWrapper)element).getTrackName());
+          }
+          else if (element instanceof NarrativeEntry)
+          {
+            layer = _tracker.getLayerFor(ImportReplay.NARRATIVE_LAYER);
+          }
+          else {
+            layer= null;
+          }
+          if(layer!=null) {
+            _layers.removeThisEditable(null, element);
+          }
+        }
+      }
+      _layers.fireExtended();
+      addedElements.clear();
+      return Status.OK_STATUS;
+    }
+  }
+
+  private boolean isContentImportable(final String content) {
+
+    boolean proceed=false;
+    StringTokenizer tokens = new StringTokenizer(content,"\\r?\\n");
+    int lineCount = 0;
+    while(tokens.hasMoreTokens() && lineCount<=6) {
+      String line = tokens.nextToken().trim();
+      if(line.startsWith(";") && !line.startsWith(";;")) {
+        StringTokenizer lineTokens = new StringTokenizer(line);
+        if(lineTokens.hasMoreTokens()) {
+          String firstWord = lineTokens.nextToken();
+          switch(firstWord) {
+            case ";CIRCLE:":

```

Comment:
@ayeshare - it will be a burden to maintain this list as we add new items.  Could you create a fancy regexp that looks for `;`, then up to _40?_ non white-space chars, then `:` ?

Reply:
this is done